internal/dag/builder_test.go

```diff
@@ -4620,6 +4700,54 @@ func TestDAGInsert(t *testing.T) {
 				},
 			),
 		},
+		"insert httpproxy with invalid tcpproxy": {
+			objs: []interface{}{proxy37, s1},
+			want: listeners(),
+		},
+		"insert httpproxy w/ tcpproxy w/ missing include": {
+			objs: []interface{}{proxy38, s1},
+			want: listeners(),
+		},
+		"insert httpproxy w/ tcpproxy w/ includes another root": {
```

Comment:
Nit: Add explanatory comment? I didn't connect all the dots until I read the `status_test.go` changes.